mlflow/models/docker_utils.py

```diff
@@ -30,106 +22,112 @@
-    )
+{setup_python_venv}
 
-    if parsed_http_proxy.username is None or parsed_http_proxy.password is None:
-        return " ".join(maven_proxy_options)
+{setup_java}
 
-    return " ".join(
-        (
-            *maven_proxy_options,
-            f"-Dhttp.proxyUser={parsed_http_proxy.username}",
-            f"-Dhttp.proxyPassword={parsed_http_proxy.password}",
-        )
-    )
-
-
-_DOCKERFILE_TEMPLATE = """
-# Build an image that can serve mlflow models.
-FROM ubuntu:20.04
-
-RUN apt-get -y update
-RUN DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC apt-get install -y --no-install-recommends \
-         wget \
-         curl \
-         nginx \
-         ca-certificates \
-         bzip2 \
-         build-essential \
-         cmake \
-         openjdk-8-jdk \
-         git-core \
-         maven \
-    && rm -rf /var/lib/apt/lists/*
-
-{setup_python_env}
-
-ENV JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
-ENV GUNICORN_CMD_ARGS="--timeout 60 -k gevent"
-# Set up the program in the image
 WORKDIR /opt/mlflow
 
 {install_mlflow}
 
-{custom_setup_steps}
+{install_model_and_deps}
 
 ENV MLFLOW_DISABLE_ENV_CREATION={disable_env_creation}
 ENV ENABLE_MLSERVER={enable_mlserver}
-
+ENV GUNICORN_CMD_ARGS="--timeout 60 -k gevent"
 
 # granting read/write access and conditional execution authority to all child directories
 # and files to allow for deployment to AWS Sagemaker Serverless Endpoints
 # (see https://docs.aws.amazon.com/sagemaker/latest/dg/serverless-endpoints.html)
 RUN chmod o+rwX /opt/mlflow/
 
-{entrypoint}
+# clean up apt cache to reduce image size
+RUN rm -rf /var/lib/apt/lists/*
+
+ENTRYPOINT ["python", "-c", "{entrypoint}"]
 """
 
 
-def _get_mlflow_install_step(dockerfile_context_dir, mlflow_home):
+SETUP_MINICONDA = """# Setup miniconda
+RUN curl -L https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh >> miniconda.sh
+RUN bash ./miniconda.sh -b -p /miniconda && rm ./miniconda.sh
+ENV PATH="/miniconda/bin:$PATH"
+"""
+
+
+def generate_dockerfile(
+    output_dir: str,
+    base_image: str,
+    model_install_steps: Optional[str],
+    entrypoint: str,
+    env_manager: Union[em.CONDA, em.LOCAL, em.VIRTUALENV],
+    mlflow_home: Optional[str] = None,
+    enable_mlserver: bool = False,
+    disable_env_creation_at_runtime: bool = True,
+):
     """
-    Get docker build commands for installing MLflow given a Docker context dir and optional source
-    directory
+    Generates a Dockerfile that can be used to build a docker image, that serves ML model
+    stored and tracked in MLflow.
     """
-    mlflow_home = os.path.abspath(mlflow_home) if mlflow_home else None
+    if base_image.startswith("python:"):
+        setup_python_venv_steps = (
+            "RUN apt-get -y update && apt-get install -y --no-install-recommends nginx"
+        )
+        setup_java_steps = ""
+        install_mlflow_steps = _pip_mlflow_install_step(output_dir, mlflow_home)
+
+    elif base_image == "ubuntu:20.04":
```

Comment:
Do we need this to be hard-coded? Can this just check against the constant `UBUNTU_BASE_IMAGE` ?